struct_test.go

```diff
@@ -472,6 +472,24 @@ func TestRegisterFunc_structArgs(t *testing.T) {
 			t.Fatalf("TakeGoUintAndReturn returned %+v wanted %+v", ret, expected)
 		}
 	}
+	{
+		// Regression test for bool field at non-zero offset
+		// Before fix: bool was always placed at bit 0 regardless of offset
+		// After fix: bool is correctly shifted to its field offset
+		type FloatAndBool struct {
+			Value     float32
+			Has_value bool
+		}
+		var FloatAndBoolFn func(FloatAndBool) int32
+		purego.RegisterLibFunc(&FloatAndBoolFn, lib, "FloatAndBool")
+		if ret := FloatAndBoolFn(FloatAndBool{Value: 12345.0, Has_value: true}); ret != 1 {
+			t.Fatalf("FloatAndBool returned %d wanted 1 (bool field not passed correctly)", ret)
```

Comment:
`FloatAndBool(true) returned %d wanted 1` don't include parentheses part since it's implicit in these tests